src/lib/input-validation/validateDomain.ts

```diff
@@ -9,21 +9,22 @@ import { Enum, ExtendEnum } from "../../types/types";
 import { CommonValidationErrorType } from "./constant/errorType";
 import { validateLength, ValidateLengthErrorType } from "./validateLength";
 
-export const ValidateDomainErrorType = ExtendEnum(
-  ValidateLengthErrorType,
-  CommonValidationErrorType.invalidFormat
-);
+export const ValidateDomainErrorType = ExtendEnum(ValidateLengthErrorType, CommonValidationErrorType.invalidFormat);
 export type ValidateDomainErrorType = Enum<typeof ValidateDomainErrorType>;
 
 export interface ValidateDomainOptions {
   maxLength?: number;
   isRequired?: boolean;
+  isSubdomainOnly?: boolean;
 }
 
 export const validateDomain = (
   value: string,
-  { maxLength = 63, isRequired = true }: ValidateDomainOptions = {}) => {
-  const REGEX = "^\\w[\\w.-]+\\w$";
+  { maxLength = 63, isRequired = true, isSubdomainOnly = false }: ValidateDomainOptions = {}
```

Comment:
Jus a note, setting the default value of `isDomainOnly` to `false` will introduce breaking change.

Reply:
updated to true

Reply:
@rayhan-ab  @aris-fm  it's ok for breaking change, the isSubdomainOnly should be false at default because it's a validateDomain.
and yes, please adjust the usage in AP, shouldn't be a lot of work IMO.

Reply:
Yes, it won't take a lot of work to adjust. I agree that `isSubdomainOnly` defaulted to `false`

Reply:
updated to false